src/Lexer.ts

```diff
@@ -0,0 +1,98 @@
+// @author Anush kumar N <anush.kumar@codingmart.com>
+
+import TokenType from './Token';
+
+type arrayOfString = Array<string>;
+type literalWithType = { type: Symbol; literal: string | undefined };
+type identifyLiteralReturnType = { token: literalWithType; updatedPos: number | undefined };
+
+// Performs lexical analysis on raw string input.
+function* Lexer(input: string) {
+  const inputAsArr: arrayOfString = Array.from(input);
+  inputAsArr.push(TokenType.eof.toString());
+
+  let position: number = 0;
+  for (; position < inputAsArr.length; position++) {
+    const str: string = inputAsArr[position];
+    const { token, updatedPos } = identifyLiteral({ str, position, inputAsArr });
+    updatedPos && (position = updatedPos);
+
+    yield token;
+  }
+}
+
+// identifyLiteral helps to identify literals i.e either the token, number or identifier
+type identifyLiteralParams = {
+  str: string;
+  position: number;
+  inputAsArr: arrayOfString;
+};
+const identifyLiteral = (data: identifyLiteralParams): identifyLiteralReturnType => {
```

Comment:
Why not just accept parameters directly? Why does it need to be a type-aliased object?

Reply:
type script throws `Binding element 'inputAsArr' implicitly has an 'any' type.` error, which forces me to type-aliased param.